nextcord/application_command.py

```diff
@@ -209,6 +211,9 @@ def __init__(self, parameter: Parameter):
         # self.required = cmd_arg.required or MISSING
         # will cause self.required to be MISSING, not False.
         self.required = cmd_arg.required if cmd_arg.required is not MISSING else MISSING
+        # Set required to False if an Optional[...] or Union[..., None] type annotation is given.
+        if type(None) in typing.get_args(parameter.annotation):
+            self.required = False
```

Comment:
What the `required` kwarg is manually set to (by the user) should always override this.
I recommend checking if `cmd_arg.required` is `MISSING` before checking the typing args.

Reply:
I think it could go either way since there's no reason to annotate a parameter as Optional if the argument itself is actually required.

I can easily swap them though.